src/main/java/org/osiam/storage/query/FilterExpression.java

```diff
@@ -23,24 +23,26 @@
 
 package org.osiam.storage.query;
 
+import org.osiam.storage.entities.ResourceEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Locale;
+
 /**
  * Representation of scim search expression.
  */
-public class ScimExpression {
+public abstract class FilterExpression<T extends ResourceEntity> {
```

Comment:
You made this class generic, but there are many places that don't have been updated to reflect this. Please review the use sites of the class.